tests/telemetry/test_telemetry.py

```diff
@@ -176,7 +176,9 @@ def test_telemetry_queue_buffer_cnt(duthosts, enum_rand_one_per_hwsku_hostname,
                   "Number of queue counters count differs from expected")
 
 
-def test_osbuild_version(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost, gnxi_path):
+@pytest.mark.parametrize('setup_streaming_telemetry', [False], indirect=True)
+def test_osbuild_version(duthosts, enum_rand_one_per_hwsku_hostname, ptfhost,
```

Comment:
I don't understand why we need to specify setup streaming telemetry fixture for all tests within test_telemetry. In conftest.py, setup_streaming_telemetry is specified to be a module level fixture so it should be applied to all tests within test_telemetry or test_events or test_telemetry_cert_rotation.

From pytest docs: https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session

We can add a scope="module" parameter to only be invoked once per test module (the default is to invoke once per test function). Multiple test functions in a test module will thus each receive the same fixture instance

Reply:
If you look at test_events.py under telemetry. All test cases have setup_streaming_telemetry passed in for each test function. I would believe this is the proper way.

Reply:
On chassis, by default the telemetry running inside gnmi container is this: /usr/sbin/telemetry -logtostderr --noTLS --port 8080 --allow_no_client_auth -v=2 --threshold 100 --idle_conn_duration 5. 
It only accepts noTLS at this moment. Need to trigger setup_streaming_telemetry to bring gnmi config for the certs and allow no client auth. Then restart gnmi will work.

If we don't put setup_streaming_telemetry for any of the test functions under test_telemetry.py for example, run whole or any one of the test functions will not trigger setup_streaming_telemetry fixture. You have to put into at least one of the function (we already have this for test_telemetry_ouput in master). But if you run only one or few test functions not including  test_telemetry_ouput the test will fail on certification error. Wrong version number here means you cannot use tls to do gnmi connection. 

          {"changed": true, "cmd": "python /root/gnxi/gnmi_cli_py/py_gnmicli.py -g -t 10.250.6.231 -p 8080 -m subscribe -x COUNTERS/Ethernet0 -xt COUNTERS_DB -o ndastreamingservertest --subscribe_mode 0 --submode 2 --interval 0 --update_count 3 ", "delta": "0:00:00.597360", "end": "2024-08-21 18:31:39.869213", "failed": true, "msg": "non-zero return code", "rc": 1, "start": "2024-08-21 18:31:39.271853", "stderr": "Traceback (most recent call last):\n  File \"/root/gnxi/gnmi_cli_py/py_gnmicli.py\", line 643, in <module>\n    main()\n  File \"/root/gnxi/gnmi_cli_py/py_gnmicli.py\", line 568, in main\n    creds = _build_creds(target, port, get_cert, certs, notls)\n  File \"/root/gnxi/gnmi_cli_py/py_gnmicli.py\", line 413, in _build_creds\n    rcert = ssl.get_server_certificate((target, port)).encode('utf-8')\n  File \"/usr/lib/python2.7/ssl.py\", line 1005, in get_server_certificate\n    with closing(context.wrap_socket(sock)) as sslsock:\n  File \"/usr/lib/python2.7/ssl.py\", line 369, in wrap_socket\n    _context=self)\n  File \"/usr/lib/python2.7/ssl.py\", line 599, in __init__\n    self.do_handshake()\n  File \"/usr/lib/python2.7/ssl.py\", line 828, in do_handshake\n    self._sslobj.do_handshake()\nssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:727)", "stderr_lines": ["Traceback (most recent call last):", "  File \"/root/gnxi/gnmi_cli_py/py_gnmicli.py\", line 643, in <module>", "    main()", "  File \"/root/gnxi/gnmi_cli_py/py_gnmicli.py\", line 568, in main", "    creds = _build_creds(target, port, get_cert, certs, notls)", "  File \"/root/gnxi/gnmi_cli_py/py_gnmicli.py\", line 413, in _build_creds", "    rcert = ssl.get_server_certificate((target, port)).encode('utf-8')", "  File \"/usr/lib/python2.7/ssl.py\", line 1005, in get_server_certificate", "    with closing(context.wrap_socket(sock)) as sslsock:", "  File \"/usr/lib/python2.7/ssl.py\", line 369, in wrap_socket", "    _context=self)", "  File \"/usr/lib/python2.7/ssl.py\", line 599, in __init__", "    self.do_handshake()", "  File \"/usr/lib/python2.7/ssl.py\", line 828, in do_handshake", "    self._sslobj.do_handshake()", "ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:727)"], "stdout": "", "stdout_lines": []}

Reply:
I might be able to set autouse=true for setup_streaming_telemetry so it's automatically used when running test. I will try that and if that works I can do without referencing it for every test function.